GrandCentralBoard/Widgets/Bonus/View/BonusWidgetView.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  Created by Krzysztof Werys on 24/02/16.
+//  Copyright © 2016 Krzysztof Werys. All rights reserved.
+//
+
+import UIKit
+import SpriteKit
+import GrandCentralBoardCore
+
+final class BonusWidgetView: UIView {
+    
+    typealias ViewModel = BonusWidgetViewModel
+    
+    @IBOutlet private weak var bonusView: SKView!
+    @IBOutlet private weak var activityView: UIView!
+    @IBOutlet private weak var activityIndicator: UIActivityIndicatorView!
+    private var scene: BonusScene!
+    
+    private(set) var state: RenderingState<ViewModel> = .Waiting {
+        didSet { handleTransitionFromState(oldValue, toState: state) }
+    }
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        
+        guard let scene = BonusScene(fileNamed:"BonusScene") else { return }
+        
+        self.scene = scene
+        scene.scaleMode = .AspectFit
+    }
+    
+    func render(viewModel: ViewModel) {
+        state = .Rendering(viewModel)
+    }
+
+    func failure() {
+        state = .Failed
+    }
+    
+    func handleTransitionFromState(state: RenderingState<ViewModel>?, toState: RenderingState<ViewModel>) {
```

Comment:
Maybe change `state: RenderingState<ViewModel>?` -> `state: RenderingState<ViewModel>` as `var state` is not an optional.